libs/domains/clusters/feature/src/lib/cluster-action-toolbar/cluster-action-toolbar.tsx

```diff
@@ -140,6 +143,28 @@ function MenuOtherActions({
 
   const canDelete = clusterStatus.status && isDeleteAvailable(clusterStatus.status)
 
+  const mutationInstallationGuide = () =>
+    openModal({
+      content: (
+        <ClusterInstallationGuideModal
+          organizationId={organizationId}
+          clusterId={cluster.id}
+          onClose={() => {
+            searchParams.delete(showSelfManagedGuideKey)
+            setSearchParams(searchParams)
+            closeModal()
+          }}
+        />
+      ),
+    })
+
+  useEffect(() => {
+    if (searchParams.has(showSelfManagedGuideKey) && cluster.kubernetes === 'SELF_MANAGED') {
+      mutationInstallationGuide()
+    }
+    return () => closeModal()
+  }, [searchParams, setSearchParams, cluster.kubernetes])
```

Comment:
```suggestion
  }, [searchParams, setSearchParams, cluster.kubernetes, mutationInstallationGuide, closeModal])
```

Reply:
I've only added `closeModal` because any constant function should be wrapped inside `useCallback` otherwise.
I don't see benefit of doing it except complying with a subjective linter rule.
https://github.com/Qovery/console/pull/1117/commits/8c12f75324a075546e51423cd514dc6d6ba909f5
https://github.com/Qovery/console/pull/1117/commits/1ae3644586cc1c3a21dca89d67e3b5778218ee34